src/sensors/fsp201sensor.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+    SlimeVR Code is placed under the MIT license
+    Copyright (c) 2022 Eiren Rain & SlimeVR contributors
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+*/
+
+#include "sensors/fsp201sensor.h"
+#include "network/network.h"
+#include "utils.h"
+#include "GlobalVars.h"
+
+void FSP201Sensor::motionSetup()
+{
+#ifdef DEBUG_SENSOR
+    imu.enableDebugging(Serial);
+#endif
+    if(!imu.begin(addr, Wire, m_IntPin)) {
+        m_Logger.fatal("Can't connect to %s at address 0x%02x", getIMUNameByType(sensorType), addr);
+        ledManager.pattern(50, 50, 200);
+        return;
+    }
+
+    m_Logger.info("Connected to %s on 0x%02x. "
+                  "Info: SW Version Major: 0x%02x "
+                  "SW Version Minor: 0x%02x "
+                  "SW Part Number: 0x%02x "
+                  "SW Build Number: 0x%02x "
+                  "SW Version Patch: 0x%02x", 
+                  getIMUNameByType(sensorType), 
+                  addr, 
+                  imu.swMajor, 
+                  imu.swMinor, 
+                  imu.swPartNumber, 
+                  imu.swBuildNumber, 
+                  imu.swVersionPatch
+                );
+
+#if BNO_USE_ARVR_STABILIZATION
+    imu.enableARVRStabilizedGameRotationVector(10);
+#else
+    imu.enableGameRotationVector(10);
+#endif
+
+#if ENABLE_INSPECTION
+    imu.enableRawGyro(10);
+    imu.enableRawAccelerometer(10);
+    imu.enableRawMagnetometer(10);
+#endif
+
+    lastReset = 0;
+    lastData = millis();
+    working = true;
+    configured = true;
+}
+
+void FSP201Sensor::motionLoop()
+{
```

Comment:
Please decide on how you're gonna format this (I usually just use the default C++ formatter from VSCode that comes with the C/C++ extension)